build-system/tasks/presubmit-checks.js

```diff
@@ -64,6 +56,18 @@ var forbiddenTerms = {
   '\\.endsWith': es6polyfill,
 };
 
+var ThreePTermsMessage = 'The 3p bootstrap iframe has no polyfills loaded and' +
+    ' can thus not use most modern web APIs.';
+
+var forbidden3pTerms = {
+  // We need to forbid promise usage because we don't have our own polyfill
+  // available. This whitelisting of callNext is a major hack to allow one
+  // usage in babel's external helpers that is in a code path that we do
+  // not use.
+  '\\.then\\((?!callNext)': ThreePTermsMessage,
```

Comment:
so we can `probably` generate a babelhelpers file that wouldn't include this piece of code. might also trim the babelhelpers file. i tried doing it originally but settled on not doing it as we were still using a lot of stuff heavily. I'll look into it again. (https://babeljs.io/docs/advanced/external-helpers/ , see --whitelist and selective builds)